doc/authentication.md

```diff
@@ -0,0 +1,47 @@
+
+Cockpit Authentication
+================================
+
+Cockpit authorizes users by looking at the Authorization header. The contents
+of that header are sent to an external command that is then responsible to:
+
+ * verify the given credentials
+ * setup an appropriate session and enviroment based on those credentials
+ * launch a bridge that can communicate with cockpit-ws on stdin and stdout.
+
+Cockpit ships with the cockpit-session command that is able to handle basic and gssapi
+authentication. Cockpit supports per auth scheme configuration so that alternate commands
+can be used.
+
+To configure an auth scheme add a section to cockpit.conf for that scheme. For example
+
+```
+[bearer]
+command = example-verify-token
+base64-decode = false
```

Comment:
Can we make this represent the entire calling convention. Right now that would be sending the rhost as the second argument, auth type as first, and decoding base64 (optionally).

```
type = plain-v1           # The default
type = base64-v1       # What cockpit-session uses
```